src/IECoreMaya/FromMayaNumericPlugConverter.cpp

```diff
@@ -59,9 +64,39 @@ FromMayaNumericPlugConverter<F,T>::FromMayaNumericPlugConverter( const MPlug &pl
 template<typename F, typename T>
 IECore::ObjectPtr FromMayaNumericPlugConverter<F,T>::doConversion( IECore::ConstCompoundObjectPtr operands ) const
 {
-	F v;
-	plug().getValue( v );
-	return new T( (typename T::ValueType)v );
+	if (plug().isArray())
+	{
+		int numElements = plug().numElements();
+		MIntArray indiceList;
+		typedef TypedData<std::vector<T> > VecDataType;
+		typename VecDataType::Ptr data = new VecDataType();
+		IntVectorData::Ptr indices = new IntVectorData();
+
+		auto &dataWritable = data->writable();
+		auto &indicesWritable = indices->writable();
+		dataWritable.resize( numElements );
```

Comment:
I think `reserve` and `push_back` is better, because `resize` is going to initialize everything as well (unless this behaviour changed in c++11).